sdk/go/common/util/gitutil/git.go

```diff
@@ -857,6 +862,52 @@ func GitListBranchesAndTags(url string) ([]plumbing.ReferenceName, error) {
 	return results, nil
 }
 
+// GetLatestTagOrHash returns the latest tag or hash in the repository.
+// To do this, we list all the tags in the repository and try to pars them as semver.  If we can't
+// find any valid tags that are semver, we'll use the hash of the default branch.  If we can't find
+// any tags or default branches we return an error.
+func GetLatestTagOrHash(ctx context.Context, url string) (*semver.Version, error) {
+	refs, err := gitListRefs(ctx, url)
+	if err != nil {
+		return nil, err
+	}
+
+	var versions []semver.Version
+	namedHashes := make(map[plumbing.ReferenceName]plumbing.Hash)
+	var headRef plumbing.ReferenceName
+	for _, ref := range refs {
+		name := ref.Name()
+		if name.IsTag() {
+			tag := name.Short()
+			v, err := semver.ParseTolerant(tag)
```

Comment:
Will this parse pre-release versions? I suspect we want to ignore them if stable versions are found, but maybe use them if no stable versions are available?

Reply:
Yeah I think it would if someone had a version like `vx.x.x-alpha` or something.  I guess we can exclude them if a stable version is found :+1: